andela/travel_tool_front
# #160370764 Search functionality

#### What does this PR do?
Add feature to enable search across all pages using the search box

#### Description of Task to be completed?
- search should be functional across all pages and filter by owner, origin, destination, status and request id

#### How should this be manually tested?
- Clone this repo - `git clone https://github.com/andela/travel_tool_front.git`
- Switch to the repo directory - `cd travel_tool_front`
- Check out this branch - `git checkout ft-search-functionality-160370764`
- Install dependencies - `yarn install`
- Start the server  - `make start`
- Set up the server by following this [guide](https://github.com/andela/travel_tool_back/pull/55)
- Navigate to `localhost:3000` in a browser
- Create a few requests
- Enter any keyword in the search box provided in the navigation bar to filter the new requests you have created
- Your requests should be filtered by the keyword you have entered in the search box
- As a manager, navigate to the approvals page
- Enter any keyword in the search box provided in the navigation bar to filter the approvals you have
- Your approvals should be filtered by the keyword you have entered in the search box


#### Any background context you want to provide?
- Fields that can be searched for are: `request ID`, `request owner`, `destination`, `origin`, `start date`, `trip type` and `request status`
- external package lodash was used to implement this feature

#### What are the relevant pivotal tracker stories?
[#160370764](https://www.pivotaltracker.com/story/show/160370764)

#### Screenshots (if appropriate)
<img width="1439" alt="screen shot 2018-09-24 at 10 10 47 am" src="https://user-images.githubusercontent.com/13545380/45944481-348d7500-bfe2-11e8-8897-5d861af7b063.png">
<img width="1440" alt="screen shot 2018-09-24 at 10 11 04 am" src="https://user-images.githubusercontent.com/13545380/45944482-348d7500-bfe2-11e8-82ed-11823d84f5bd.png">
<img width="1440" alt="screen shot 2018-09-24 at 11 28 15 am" src="https://user-images.githubusercontent.com/13545380/45947691-19743280-bfed-11e8-9702-6ce316c4e793.png">



#### Questions:
